src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -535,26 +534,18 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
 
     void visit_ConstantInteger(const ASR::ConstantInteger_t &x) {
         src = std::to_string(x.m_n);
-        last_unary_plus = false;
-        last_binary_plus = false;
```

Comment:
I think you have to set the last precedence for each such place that you removed `last_unary_plus`. Essentially you have to go into each expression node and ensure the last precedence is set. That should fix the issue with extra (inconsistent) parentheses.